webapp/app/js/compare.js

```diff
@@ -0,0 +1,448 @@
+                          { project: $scope.newProject,
+                            signature: new_seriesList[idx].signature,
+                            visible: true}));
+
+            var detailsLink = thServiceDomain + '/perf.html#/graphs?timerange=' +
+                $scope.timeRange + '&series=' + newSeries;
+
+            if (seriesList[idx].signature != new_seriesList[idx].signature) {
+              detailsLink += '&series=' + originalSeries;
+            }
+
+            detailsLink += '&highlightedRevision=' + $scope.newRevision;
+
+            cmap.detailsLink = detailsLink;
+            cmap.name = seriesList[idx].platform;
+            cmap.isHeader = false;
+            compareResultsMap[counter++] = cmap;
+          }
+
+        };
+      });
+      $scope.compareResults = Object.keys(compareResultsMap).map(function(k) {
+        return compareResultsMap[k];
+      });
+    }
+
+    function verifyRevision(project, revision, rsid) {
+      if ((revision != null && revision != '') &&
+          (project != null && project != '')) {
+
+        var uri = thServiceDomain + '/api/project/' + project +
+            '/resultset/?format=json&full=false&with_jobs=false&revision=' + 
+            revision;
+
+        $http.get(uri).then(function(response) {
+          var results = response.data.results;
+          if (results.length > 0) {
+
+            //TODO: remove this hack so we can return the value
+            if (rsid == 'original') {
+              $scope.originalResultSetID = results[0].id;
+            } else {
+              $scope.newResultSetID = results[0].id;
+            }
+          }
+        });
+      }
+    }
+
+    function updateURL() {
+      $state.transitionTo('compare', { 'originalProject': $scope.originalProject,
+                            'originalRevision': $scope.originalRevision,
+                            'newProject': $scope.newProject,
+                            'newRevision': $scope.newRevision},
+                {location: true, inherit: true, notify: false, relative: $state.$current});
+    }
+
+    var optionCollectionMap = {};
+
+    $http.get(thServiceDomain + '/api/optioncollectionhash').then(
+      function(response) {
+        response.data.forEach(function(dict) {
+          optionCollectionMap[dict.option_collection_hash] =
+            dict.options.map(function(option) {
+              return option.name; }).join(" ");
+        });
+      }).then(function() {
+        if ($stateParams.e10s) {
+          $stateParams.e10s = true;
+        } else {
+          $stateParams.e10s = false;
+        }
+
+        if ($stateParams.pgo) {
+          $stateParams.pgo = true;
+        } else {
+          $stateParams.pgo = false;
+        }
+
+        // TODO: validate projects and revisions
+        $scope.originalProject = '';
+        if ($stateParams.originalProject) {
+          $scope.originalProject = $stateParams.originalProject;
+        }
```

Comment:
I don't understand the point of all this if stuff. Why not just say $scope.originalProject = $stateParams.originalProject; ?